### PR TITLE
solution 01/12

### PR DESCRIPTION
**Pull request by VadimKadasiev**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    void get_nullLogin_notOk() {
+        assertNull(storageDao.get(null));
+    }
+
+    @Test
+    void get_notExistedLogin_notOk() {
+        assertNull(storageDao.get("login"));
+    }
+
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:48]: remove all dao tests, we shouldn't test dao here, only registrationService

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        user.setAge(23);
+        user.setPassword("validPassword");
+        user.setLogin("validLogin");
+    }
+
+    @Test
+    void addAndGetValidUserDirectlyInStorage_Ok() {
+        Storage.people.add(user);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:31]: in  RegistrationServiceImplTest we only test RegistrationServiceImpl

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Test
+    void register_shortLogin_notOk() {
+        user.setLogin("");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        user.setLogin("abc");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        user.setLogin("abcde");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:70]: good job, thorough testing 👏

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    void get_notExistedLogin_notOk() {
+        assertNull(storageDao.get("login"));
+    }
+
+    @Test
+    void register_ExistedLogin_notOk() {
+        User user1 = user;
+        registrationService.register(user);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:53]: image

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final User user = new User();
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:21]: clear in @AfterEach

**Review by okuzan — CHANGES_REQUESTED** [review]
always check common mistakes before requesting a review (found at the end of task description)
https://github.com/mate-academy/jv-registration-validation/blob/master/checklist.md

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 6]
```diff
@@ -5,10 +5,37 @@
...
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private static final String SHORT_LENGTH_MESSAGE
+            = " shouldn't contain less than six characters";
+    private static final String NULL_DATA_MESSAGE = " can't be null";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:16]: what if user is null?

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java (hunk 6 above)
Comment (VadimKadasiev) [reply src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:16]: Oops... 
Got it!

**Review by kozhukhovsky — APPROVED** [review]
Good job!

**Pull request closed without merging**